python/ray/data/_internal/execution/operators/actor_pool_map_operator.py

```diff
@@ -249,6 +250,32 @@ def shutdown(self):
         self._actor_pool.kill_all_actors()
         super().shutdown()
 
+        # Warn if the user specified a batch or block size that prevents full
+        # parallelization across the actor pool. We only know this information after
+        # execution has completed.
+        total_rows = sum([m.num_rows for m in self._output_metadata])
+        min_workers = self._autoscaling_policy.min_workers
+        max_desired_batch_size = total_rows // min_workers
+        if (
+            self._min_rows_per_bundle is not None
+            and self._min_rows_per_bundle > max_desired_batch_size
+        ):
+            # The user specified a batch size, but it was probably too large.
+            logger.get_logger().warning(
+                "To ensure full parallelization across an actor pool of size "
+                f"{min_workers}, the specified batch size "
+                f"should be at most {max_desired_batch_size}. Your configured batch "
+                f"size for this operator was {self._min_rows_per_bundle}."
+            )
+        elif len(self._output_metadata) < min_workers:
```

Comment:
even if there are a small number of blocks, if the batch size is small, this can still be parallelized across multiple actors?

Reply:
I don't believe we ever auto split blocks due to batch size (only ever coalesce blocks up to the min batch size if the user explicitly specifies it).

Reply:
Oh I see...I thought it was using the same mechanism as for `iter_batches`